### DISCUSSION TITLE
Using the value of a relationship fieldtype in collection routes

### DISCUSSION THREAD

**Discussion by duncanmcclean** [statamic/cms#6650] [Troubleshooting] [answered]
We have a custom Relationship fieldtype which we'd like to use in our collection route like so but it doesn't seem to work...

```yaml
route: '/locations/{location:slug}/apartment-types/{slug}'
```

But this does work (without the `:slug` bit)

```yaml
route: '/locations/{location}/apartment-types/{slug}'
```

We are returning an augmented array from our custom Location fieldtype - it returns the augmented array from an entry. 

Is there any way to workaround the issue we're running into (without having to build custom routes?)

**Comment by hlorofos** [comment]
I wasn't been able to find the solution other that custom template which retrieves the entries from the collections by their slugs.

**Reply by duncanmcclean** [reply, marked as answer]
You can workaround this by creating a [Computed Value](https://statamic.dev/computed-values#on-entry-instances) to return any related fields you want to use in your collection route.

For example, you could do something like this:

```php
Collection::computed('locations', 'location_slug', function ($entry, $value) {
    return $entry->entries_field->slug();
});
```

Then, the collection's route would use that computed value like this: `/locations/{location_slug}/{slug}`.

If you want this to be supported in future version of Statamic, add a 👍 reaction to this feature request: statamic/ideas#360

**Reply by hlorofos** [reply]
Oh, that's I haven't realized there is such a way. I'll give it a try. Feature request is also handy, upvoted!
Thank you very much for the hint

**Reply by duncanmcclean** [reply]
No worries! Hope it works - have a great rest of your day.
